source/common/upstream/cluster_manager_impl.cc

```diff
@@ -2399,6 +2399,20 @@ Http::ConnectionPool::InstancePtr ProdClusterManagerFactory::allocateConnPool(
   }
   if (protocols.size() == 1 && protocols[0] == Http::Protocol::Http2 &&
       context_.runtime().snapshot().featureEnabled("upstream.use_http2", 100)) {
+    if (host->cluster().clusterType().has_value() &&
+        host->cluster().clusterType()->name() == "envoy.clusters.reverse_connection") {
+      ENVOY_LOG(debug, "allocating reverse connection pool");
+      auto* factory = Config::Utility::getFactory<Http::Http2::ReverseConnPoolFactory>(
+          envoy::extensions::upstreams::http::reverse_conn::v3::ReverseConnPoolProto::
```

Comment:
Rather than a default instance, you would need to pass the `ClusterType()` to `getFactory` or `ClusterType()->typed_config()` to `getFactoryByType()`, and then define a customized connection pool factory in this PR: https://github.com/envoyproxy/envoy/pull/37931. So that it is extendable and it is using the typed_config to search the factory, and no default proto is needed.